### PR TITLE
Add hobo reels

### PR DESCRIPTION
**Pull request by NoahGoldFox**
#### Summary

SUMMARY: Content "Add 2 hobo reels"

#### Purpose of change

Add a platic hobo reel based on the [Ka-bar kaster,](https://www.kabar.com/products/product.jsp?item=9921) and one based on generic wooden craftmen ones. The reels are less effective then the current rods, but make for for that by being very compact. The plastic one you need to find, and its not that common, and the wooden one takes more skills to craft then the basic rod. 

#### Describe the solution

Added the items.

Added the plastic reel to several itemgroups, and added a recipe to craft the wooden reel. You can find the recipe in some survival recipe books.

#### Testing
Spawned both reels and fished with them to make sure they worked. Crafted the wooden reel. spawned overmap specials with extra loot chance to make sure the item would spawn.

#### Additional context
This is my first PR, and i wanted to add something my style. I really like interesting gadgets and fishing in videogames, so this was a good choice. I also felt this just is a verrry survivalist item.

### CODE REVIEW

File: data/json/items/tool/fishing.json [hunk 1]
```diff
@@ -82,5 +82,46 @@
     "use_action": [ "FISH_ROD" ],
     "qualities": [ [ "FISHING", 2 ] ],
     "flags": [ "FISH_GOOD", "SHEATH_SPEAR" ]
+  },
+  {
+    "id": "plastichoboreel",
+    "type": "TOOL",
+    "looks_like": "baton",
+    "//": "based on the ka-bar kaster",
+    "name": { "str": "Plastic hand fishing reel" },
```
Comment (Qrox) [inline data/json/items/tool/fishing.json:91]:
```suggestion
    "name": { "str": "plastic hand fishing reel" },
```

File: data/json/items/tool/fishing.json [hunk 2]
```diff
@@ -82,5 +82,46 @@
...
+    "description": "A commercial version of what is often known as a 'hobo fishing reel'. Is is a short rod with a section on one end with fishing line wrapped around it, with a rotating guard around that. It is not as effective as a normal fishing pole, but is much more compact than them,",
```
Comment (Qrox) [inline data/json/items/tool/fishing.json:92]:
```suggestion
    "description": "A commercial version of what is often known as a 'hobo fishing reel'.  It is a short rod with a section on one end with fishing line wrapped around it, with a rotating guard around that.  It is not as effective as a normal fishing pole, but is much more compact than them.",
```

File: data/json/items/tool/fishing.json [hunk 3]
```diff
@@ -82,5 +82,46 @@
...
+    "use_action": [ "FISH_ROD" ],
+    "qualities": [ [ "FISHING", 1 ] ],
+    "flags": [ "FISH_GOOD" ]
+  },
+  {
+    "id": "hoboreel",
+    "type": "TOOL",
+    "looks_like": "pointy_stick",
+    "name": { "str": "hobo hand fishing reel" },
+    "description": "What is often known as a 'hobo fishing reel'. Is is a short wooden rod with a section on one end with fishing line wrapped around it, with a notch to put the string to help pull in fish. It is not as effective as a normal fishing pole, but is much more compact than them,",
```
Comment (Qrox) [inline data/json/items/tool/fishing.json:112]:
```suggestion
    "description": "What is often known as a 'hobo fishing reel'.  It is a short wooden rod with a section on one end with fishing line wrapped around it, with a notch to put the string to help pull in fish.  It is not as effective as a normal fishing pole, but is much more compact than them.",
```

File: data/json/items/tool/fishing.json [hunk 4]
```diff
@@ -82,5 +82,66 @@
...
+  "id": "hoboreel",
+  "type": "TOOL",
+  "looks_like": "pointy_stick",
+  "name": {
+    "str": "hobo hand fishing reel"
+  },
+  "description": "What is often known as a 'hobo fishing reel'.  It is a short wooden rod with a section on one end with fishing line wrapped around it, with a notch to put the string to help pull in fish.  It is not as effective as a normal fishing pole, but is much more compact than them.",
+  "weight": "500 g",
+  "volume": "380 ml",
+  "longest_side": "150 mm",
```
Comment (LaVeyanFiend) [inline data/json/items/tool/fishing.json:127]: Were the units supposed to be centimeters?

Comment (NoahGoldFox) [reply data/json/items/tool/fishing.json:127]: No, the hobo fishing reel is only about that long.

**Comment by AMurkin** [comment]
To make the summary valid it should be like so:
SUMMARY: Content "Add 2 hobo reels"

Use this tool to format your changes: http://dev.narc.ro/cataclysm/format.html.
[More info.](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_STYLE.md)

**Comment by AMurkin** [comment]
I meant that you need to format the entire contents of the file, and then the changes will be formatted correctly.

**Comment by NoahGoldFox** [comment]
But wouldnt the pull request then be saying i was changing all that code instead of just the specific stuff im adding?

**Comment by anothersimulacrum** [comment]
No, that's already linted.

**Comment by Qrox** [comment]
Copy your changed json files into the text box at http://dev.narc.ro/cataclysm/format.html, click lint, copy the linted result back, commit your changes, and push. Don't worry about the unchanged part, as it is already linted and won't change if you didn't change it.

And here is some general guideline on json formatting: When in doubt, just follow the format of the previous code (space after comma, spaces inside brackets, etc). If a line exceeds 120 characters, break it after each start bracket and comma and before each ending bracket.

**Comment by NoahGoldFox** [comment]
I did that, but its saying i changed alot more then i actually am

**Comment by NoahGoldFox** [comment]
and the style check is still failing

**Comment by NoahGoldFox** [comment]
And it just added in some stuff that wasnt there before and now it has json errors!

**Comment by Qrox** [comment]
> but its saying i changed alot more then i actually am

I'm not sure what you mean by "it". The json linter? The building matrix? The diff viewer?

> And it just added in some stuff that wasnt there before and now it has json errors!

That seems to be introduced by your last three commits. It seems you didn't replace the whole file with the linted result but inserted it into the file.

**Comment by NoahGoldFox** [comment]
damnit, that happened because i did "select all" to replace those and it didnt work right for some reason

**Comment by NoahGoldFox** [comment]
and its still not working. i put the whole file through the linter but the style check is still failing

File: data/json/itemgroups/Locations_MapExtras/locations.json [hunk 5]
```diff
@@ -1,4 +1,4 @@
-[
+[
```
Comment (Qrox) [inline data/json/itemgroups/Locations_MapExtras/locations.json:1]:
```suggestion
[
```
Stray spaces here.

File: data/json/items/tool/fishing.json [hunk 6]
```diff
@@ -82,5 +82,47 @@
...
+    "bashing": 5,
+    "material": [ "wood" ],
+    "symbol": "/",
+    "color": "brown",
+    "use_action": [ "FISH_ROD" ],
+    "qualities": [ [ "FISHING", 1 ] ],
+    "flags": [ "FISH_POOR" ]
   }
 ]
+
```
Comment (Qrox) [inline data/json/items/tool/fishing.json:128]:
```suggestion
```
Stray empty line

File: data/json/recipes/other/tool.json [hunk 7]
```diff
@@ -388,6 +388,35 @@
     "components": [ [ [ "plastic_chunk", 3 ] ], [ [ "wire", 1 ] ], [ [ "scrap", 1 ] ], [ [ "weights", 40, "LIST" ] ] ]
   },
   {
+  "type": "recipe",
```
Comment (Qrox) [inline data/json/recipes/other/tool.json:391]: The indentation is wrong.

**Comment by NoahGoldFox** [comment]
:D it finally passed the style test

**Comment by NoahGoldFox** [comment]
I couldnt find why it didnt pass the clang-tidy check

**Comment by Qrox** [comment]
The clang-tidy check fails on the c++ code so that's most likely not related to this PR.

**Comment by NoahGoldFox** [comment]
Ah cool. then seems this finally is done and good

**Pull request merged**